repo_health_dashboard/utils/utils.py

```
@@ -184,3 +212,51 @@ def write_squashed_metadata_to_html(metadata_by_repo=None, filename="dashboard.h
         f.write("</tbody>\n")
         f.write("</table>\n")
         f.write("</body></html>\n")
+
+
+def write_squashed_metadata_to_sqlite(metadata_by_repo, table_name, configuration, sqlite_output):
+    """
+    Assume all the metadata_by_repo have the same keys
+    """
+    # Get a superset of all keys in the metadata_by_repo dictionaries
+    sorted_keys = get_sorted_keys(metadata_by_repo, configuration)
+
+    # Change key names to their aliases for display
+    sorted_aliased_keys = get_sorted_aliased_keys(metadata_by_repo, configuration)
+
+    # Connect to the SQLite database file
+    conn = sqlite3.connect(sqlite_output+".db")
+    c = conn.cursor()
+
+    columns = ["repo_name"] + sorted_aliased_keys
+
+    # Replace '.', '-', ':' with _ in the column names because sqlite doesn't accept those tokens in column names
+    columns = [re.sub(r"[\.\-\:]", "_", column) for column in columns]
+    sqlite_columns = []
+    # Get first row of the data and infer types of the columns from it
```

Hmm, this isn't ideal if the first row happens to be missing data for some of the checks.  Good enough for now, but I'd love to eventually add data type to the metadata that we specify for each check (possibly via type annotations).

---

I thought SQLite basically ignore the data types anyway?

---

We can't perform visualizations and other operations on data in metabase if all the data is text hence the need of evaluating data types.

---

Yeah, internally SQLite is pretty fuzzy with data type handling, but Metabase and other tools rely on them more.  And SQLite tries to be consistent with declared data types when spitting out results, even if it silently converts them internally.